### PR TITLE
🐛 Undo `hostname` change on Safari and re-enable testing on latest version

### LINKED ISSUE #15748: Test failures on Saucelabs Safari 11.1
## What's the issue?
Bug

## How do we reproduce the issue?
Check Travis for any incoming PR.

I see that any incoming PR post https://github.com/ampproject/amphtml/pull/15614/files is basically broken.

The tests that are broken are either from performance or a4a.

Broken tests: https://travis-ci.org/ampproject/amphtml/jobs/386826879

One of the error: 
```
Blocked a frame with origin "http://localhost:9876" from accessing a frame with origin "https://www.google.com".  The frame requesting access has a protocol of "http", the frame being accessed has a protocol of "https". Protocols must match.
```

might match something to do with `isSecureUrl`

## What browsers are affected?
None

## Which AMP version is affected?
Current

### LINKED ISSUE #15751: Tests are running on different hostnames
```
hostname: process.platform === 'darwin' ? '127.0.0.1' : 'localhost',
```

This failed some tests (test-amp-analytics.js) locally where string comparison is operated with http://localhost/ .

Having this discrepancy between environments can be a bug hole. Can we unify to 127.0.0.1 for instance?

caused by #15510

### PR DESCRIPTION
**Pull request by rsimha**
Some tests were failing on Safari because we were setting the Karma `hostname` to `127.0.0.1` instead of `localhost`. This is because we have hard-coded checks for `localhost` all over the place in our tests.

This PR undoes the change in #15510, pending a search for a different solution to the occasional Karma disconnect problem on Safari. It also re-enables testing on the latest (11.1.0) Safari version after skipping a few unit tests on Safari that fail on 11.1.0. They'll still run on Chrome.

Partial revert of #15510
Fixes #15751 
Fixes #15748
Related to #14848

### CODE REVIEW

**Comment by rsimha** [comment]
Assigning to @danielrozenberg to take over. I think there are a couple more unit test failures, and I'm not sure if all this needs is a sync / rebase.

File: ads/google/a4a/test/test-line-delimited-response-handler.js [hunk 1]
```diff
@@ -168,7 +168,9 @@ describe('#line-delimited-response-handler', () => {
       sandbox.restore();
     });
 
-    it('should handle empty streamed response properly', () => {
+    // TODO(amphtml, #15748): Fails on Safari 11.1.0.
```
Comment (choumx) [inline ads/google/a4a/test/test-line-delimited-response-handler.js]: Can we change this to the a4a owner, @ mention them here and file/assign issues?

File: test/functional/test-resources.js [hunk 2]
```diff
@@ -2820,7 +2820,9 @@ describes.fakeWin('Resources.add/upgrade/remove', {amp: true}, env => {
     });
   });
 
-  it('should not schedule pass when immediate build fails', () => {
+  // TODO(amphtml, #15748): Fails on Safari 11.1.0.
+  it.configure().skipSafari('should not schedule pass when immediate ' +
```
Comment (choumx) [inline test/functional/test-resources.js]: Ditto here. /cc @jridgewell

File: build-system/tasks/karma.conf.js [hunk 3]
```diff
@@ -42,9 +42,10 @@ module.exports = {
     './testing/**/*.js': ['browserify'],
   },
 
-  // Sauce labs on Safari doesn't support 'localhost' addresses. See #14848.
+  // TODO(rsimha, #15510): Sauce labs on Safari doesn't reliably support
+  // 'localhost' addresses. See #14848 for more info.
   // Details: https://support.saucelabs.com/hc/en-us/articles/115010079868
-  hostname: process.platform === 'darwin' ? '127.0.0.1' : 'localhost',
+  hostname: 'localhost',
```
Comment (choumx) [inline build-system/tasks/karma.conf.js]: +1

**Comment by rsimha** [comment]
@danielrozenberg ^^

File: ads/google/a4a/test/test-line-delimited-response-handler.js (hunk 1 above)
Comment (danielrozenberg) [reply ads/google/a4a/test/test-line-delimited-response-handler.js]: /to @lannka (based on go/amp-components)

**Comment by rsimha** [comment]
Thanks for merging this @danielrozenberg!

@lannka Let us know if you continue to see errors due to hostname mismatches.

We'll need to keep an eye on Safari disconnect failures after this, and work on a different fix (that doesn't involve changing the hostname) if the incidence continues to be high.

**Pull request merged**